### PR TITLE
chore(ci): use backend-native with python cached artifract for dev docker image

### PR DESCRIPTION
**Pull request by KSDaemon**

### CODE REVIEW

File: .github/workflows/rust-cubesql.yml [hunk 1]
```diff
@@ -196,6 +196,13 @@ jobs:
           CUBEJS_NATIVE_INTERNAL_DEBUG: true
         working-directory: ./packages/cubejs-testing
         run: yarn smoke:cubesql
+      - name: Store build artifact for dev image
+        if: (matrix.python-version == '3.11') && (matrix.target == 'x86_64-unknown-linux-gnu')   # No need to create many versions
+        uses: actions/upload-artifact@v4
+        with:
+          name: "dev-cubejs-backend-native-x86_64"    # this name is referenced in master.yml->docker-image-dev
+          path: ./packages/cubejs-backend-native
```
Comment (ovr) [inline .github/workflows/rust-cubesql.yml:204]: This will upload target directory. You need to upload `index.node` file

Comment (KSDaemon) [reply .github/workflows/rust-cubesql.yml]: fixed

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -37,6 +37,84 @@ jobs:
...
+        working-directory: ./packages/cubejs-testing
+        run: yarn smoke:cubesql
+      - name: Store build artifact for dev image
+        uses: actions/upload-artifact@v4
+        with:
+          name: "native-linux-x64-glibc-3.9.node"    # this name is referenced below in docker-image-dev
+          path: ./packages/cubejs-backend-native/native/index.node
+          overwrite: true
+
   docker-image-dev:
```
Comment (ovr) [inline .github/workflows/master.yml:118]: `need: [build_native_linux]`

File: .github/workflows/rust-cubesql.yml [hunk 3]
```diff
@@ -196,6 +196,13 @@ jobs:
           CUBEJS_NATIVE_INTERNAL_DEBUG: true
         working-directory: ./packages/cubejs-testing
         run: yarn smoke:cubesql
+      - name: Store build artifact for dev image
```
Comment (ovr) [inline .github/workflows/rust-cubesql.yml:199]: Then you can remove?)

File: .github/workflows/master.yml [hunk 4]
```diff
@@ -37,6 +37,84 @@ jobs:
...
+          PYO3_PYTHON: python3.9
+          CARGO_BUILD_TARGET: x86_64-unknown-linux-gnu
+        working-directory: ./packages/cubejs-backend-native
+        run: yarn run native:build-debug-python
+      - name: Test native (GNU only)
+        env:
+          CUBESQL_STREAM_MODE: true
+          CUBEJS_NATIVE_INTERNAL_DEBUG: true
+        working-directory: ./packages/cubejs-backend-native
+        run: yarn run test:unit
```
Comment (ovr) [inline .github/workflows/master.yml:105]: We don't need unit tests here.

File: .github/workflows/master.yml [hunk 5]
```diff
@@ -37,6 +37,84 @@ jobs:
...
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn cache dir)" >> "$GITHUB_OUTPUT"
+      - name: Restore yarn cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Yarn install
```
Comment (ovr) [inline .github/workflows/master.yml:82]: We dont need to install all packages, compiling tsc
let's use what we use in the publish

```
      - name: Set Yarn version
        run: yarn policies set-version v1.22.19
      - name: Install cargo-cp-artifact
        run: npm install -g cargo-cp-artifact@0.1

```

**Review by ovr — APPROVED** [review]

**Review by ovr — APPROVED** [review]

**Pull request merged**